core/modules/layout/includes/layout.class.inc

```diff
@@ -764,21 +766,90 @@ class Layout {
       }
     }
 
-    // Assign orphan blocks to the default or last region, if the former doesn't
-    // exist.
     if (!empty($this->orphaned_blocks)) {
-      end($new_positions);
-      $this->refuge_region = !empty($new_layout_info['default region']) ? $new_layout_info['default region'] : key($new_positions);
-      foreach ($this->orphaned_blocks as $uuid) {
-        $new_positions[$this->refuge_region][] = $uuid;
-      }
+      $this->setOrphanedBlockMessage();
     }
 
     $this->layout_template = $layout_name;
     $this->positions = $new_positions;
     module_invoke_all('layout_template_change', $this, $old_layout_info['name']);
   }
 
+  /**
+   * Set a Refuge Region for orphan blocks.
+   */
+  function setRefugeRegion() {
+    $template_info = layout_get_layout_template_info($this->layout_template);
+    // First check for a default region defined by the layout template.
+    if (!empty($template_info['default region'])) {
+      $this->refuge_region  = $template_info['default region'];
+    }
+    // Then check for a region named "content".
+    elseif (array_key_exists('content', $template_info['regions'])) {
+      $this->refuge_region = 'content';
+    }
+    // Finally, default to the last region defined by the layout template.
+    else {
+      end($template_info['regions']);
+      $this->refuge_region = key($template_info['regions']);
+    }
+  }
+
+  /**
+   * Set a message about orphan blocks being moved to the refuge region.
+   */
+  function setOrphanedBlockMessage() {
+    if (!empty($this->orphaned_blocks)) {
```

Comment:
It feels like this check in not necessary. If something is calling `setOrphanedBlockMessage()`, it means that they must have already checked that. Alternatively, this method should be renamed to something like `checkOrphanedBlock()`. If the latter, then we should also remove this check earlier:
```php
    if (!empty($this->orphaned_blocks)) {
      $this->setOrphanedBlockMessage();
    }
```

...and simply call `$this->checkOrphanedBlock()` instead.

Reply:
The function should only do one thing so `this->checkOrphanedBlocks()` is not clear if it's both checking if there are orphaned blocks and it's setting a message.